add Storybook (js) to recipes (#23665)
gatsbyjs/gatsby: packages/gatsby-recipes/recipes/storybook-js.mdx

<NPMPackage name="babel-plugin-remove-graphql-queries" />
<NPMPackage name="babel-plugin-react-docgen" />

---

Install Storybook React NPM packages and addons

<NPMPackage name="@storybook/react" />
<NPMPackage name="@storybook/addon-actions" />
<NPMPackage name="@storybook/addon-docs" />

---

Create JavaScript Storybook webpack config (main.js)

<File
  path="./.storybook/main.js"
  content="https://raw.githubusercontent.com/PaulieScanlon/gatsby-recipe-storybook-js/master/recipe/main.js"
/>

---

Configure Storybook / Gatsby Link settings (preview.js)

<File
  path="./.storybook/preview.js"
  content="https://raw.githubusercontent.com/PaulieScanlon/gatsby-recipe-storybook-js/master/recipe/preview.js"
/>

---

Create example JavaScript Link .stories

<File
  path="./src/components/Link.stories.js"
  content="https://raw.githubusercontent.com/PaulieScanlon/gatsby-recipe-storybook-js/master/recipe/Link.stories.js"
/>

---

Add Storybook npm scripts

<NPMScript
  name="storybook"
  command="NODE_ENV=production start-storybook -s ./public -p 6006 --ci"
/>
<NPMScript
  name="build-storybook"
  command="NODE_ENV=production build-storybook"
/>

---

## You're all done!

> Before your run Storybook make sure you run gatsby develop or gatsby build. This is so GraphQL data is available in Storybook

Reviewer: ```suggestion
Before your run Storybook make sure you run `gatsby develop` or `gatsby build`. This ensures that Gatsby's GraphQL data is available in Storybook.
```
